plugins/pipelines/test/lib/hooks_test.rb

```diff
@@ -0,0 +1,42 @@
+require_relative '../test_helper'
+
+describe 'zendesk hooks' do
+  let(:deploy) { deploys(:succeeded_test) }
+  let(:next_deploy) { deploys(:succeeded_production_test) }
+  let(:stage) { deploy.stage }
+  let(:next_stages) { [ stages(:test_production), stages(:test_production_pod) ] }
+  let(:output) { StringIO.new }
+  let(:job) do
+    Job.create(project: stage.project,
+               command: "echo hello world",
+               status: "running",
+               user: User.first,
+               deploy: deploy)
+  end
+
+
+  describe :before_execute_finish_msg do
+    it 'kicks off the next stages in the deploy' do
+      stage.update!(next_stage_ids: next_stages.map(&:id))
+      DeployService.any_instance.expects(:deploy!).returns(next_deploy).twice
+      Samson::Hooks.fire(:before_execute_finish_msg, job, true, output)
+    end
+
+    it 'does not kick off the next stage in the pipeline if current stage failed' do
+      stage.update!(next_stage_ids: next_stages.map(&:id))
+      DeployService.any_instance.expects(:new).never
+      Samson::Hooks.fire(:before_execute_finish_msg, job, false, output)
+    end
+
+    it 'does not deploy another if the next_stage_id is nil' do
+      DeployService.any_instance.expects(:new).never
+      Samson::Hooks.fire(:before_execute_finish_msg, job, true, output)
+    end
+
+    it 'does not deploy another if the deploy is nil' do
+      job = Job.create(project: stage.project, command: "echo", status: "running", user: User.first)
+      DeployService.any_instance.expects(:new).never
```

Comment:
would be nice if these expectations were symmetric like `DeployService.any_instance.expects(:deploy!).never`